packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -163,15 +206,54 @@ export class Table extends Construct {
   }
 
   public addPartitionKey(attribute: Attribute): this {
-    this.addKey(attribute.name, attribute.type, HASH_KEY_TYPE);
+    this.addKey(attribute, HASH_KEY_TYPE);
     return this;
   }
 
   public addSortKey(attribute: Attribute): this {
-    this.addKey(attribute.name, attribute.type, RANGE_KEY_TYPE);
+    this.addKey(attribute, RANGE_KEY_TYPE);
     return this;
   }
 
+  public addGlobalSecondaryIndex(props: SecondaryIndexProps) {
+    if (this.globalSecondaryIndexes.length === 5) {
```

Comment:
According to this document https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes
Number should be 20

Reply:
In the latest version there is no limit at all. https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-dynamodb/lib/table.ts